File: backend/models/bookModel.js
const db = require('../config/db');

const getAllBooks = async () => {
    const [rows] = await db.query('SELECT * FROM books ORDER BY created_at DESC');
    return rows;
};

const getBookById = async (id) => {
    const [rows] = await db.query('SELECT * FROM books WHERE id = ?', [id]);
    return rows[0];
};

const createBook = async (title, author, published_year, image, stock) => {
    const [result] = await db.query(
        'INSERT INTO books (title, author, published_year, image, stock) VALUES (?, ?, ?, ?, ?)',
        [title, author, published_year, image, stock]
    );
    return result.insertId;
};

const updateBook = async (id, title, author, published_year, image, stock) => {
    
    const [result] = await db.query(
        'UPDATE books SET title = ?, author = ?, published_year = ?, stock = ?, image = COALESCE(?, image) WHERE id = ?',
        [title, author, published_year, stock, image, id]
    );
    return result.affectedRows;
};

const deleteBook = async (id) => {
    const [result] = await db.query('DELETE FROM books WHERE id = ?', [id]);
    return result.affectedRows;
};

module.exports = { getAllBooks, getBookById, createBook, updateBook, deleteBook };

File: librasys_mobile/lib/models/book_model.dart
class Book {
  final int id;
  final String title;
  final String author;
  final int publishedYear;
  final int stock;
  final String? image;
  final DateTime? createdAt;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.publishedYear,
    required this.stock,
    this.image,
    this.createdAt,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      author: json['author'],
      publishedYear: json['published_year'],
      stock: json['stock'],
      image: json['image'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null, 
    );
  }
}